rs/system_api/src/sandbox_safe_system_state.rs

```diff
@@ -881,16 +881,17 @@ impl SandboxSafeSystemState {
         self.update_balance_change(new_balance);
     }
 
-    pub(super) fn mint_cycles(&mut self, amount_to_mint: Cycles) -> HypervisorResult<()> {
+    pub(super) fn mint_cycles(&mut self, amount_to_mint: Cycles) -> HypervisorResult<Cycles> {
+        let old_balance = self.cycles_balance();
         let mut new_balance = self.cycles_balance();
-        let result = self
-            .cycles_account_manager
+        self.cycles_account_manager
```

Comment:
Could we let the cycles account manager compute the difference? Otherwise, it computes an inconsistent value that is silently discarded (and could be activated again after a refactoring in the future).

Reply:
The return type of this is (), so idk what would be inconsistent.

Reply:
I see. Then nevermind the "Otherwise, ...". Still, is there a reason to not do in the cycles account manager?